modules/core/src/Auth/Source/SourceIPSelector.php

```diff
@@ -91,18 +90,18 @@ public function __construct(array $info, array $config)
      */
     protected function selectAuthSource(/** @scrutinizer ignore-unused */ array &$state): string
     {
-        $netUtils = new Utils\Net();
-        $ip = $_SERVER['REMOTE_ADDR'];
+        $ip = Request::createFromGlobals()->getClientIp();
+        Assert::notNull($ip, "Unable to determine client IP.");
 
         $source = $this->defaultSource;
         foreach ($this->zones as $name => $zone) {
             foreach ($zone['subnet'] as $subnet) {
-                if ($netUtils->ipCIDRcheck($subnet, $ip)) {
+                if (IpUtils::checkIp($ip, $subnet)) {
                     // Client's IP is in one of the ranges for the secondary auth source
                     Logger::info(sprintf(
                         "core:SourceIPSelector:  Selecting zone `%s` based on client IP %s",
                         $name,
-                        $ip
+                        $ip,
```

Comment:
small thing, both this and https://github.com/simplesamlphp/simplesamlphp/pull/2087/files have a trailing comma after the $ip. It is handy that php -a doesnt seem to care about this and maybe it is not so much of an issue for sprintf()s that might have trailing args anyway.

Reply:
Trailing comma's are perfectly fine in PHP since 7.2. It reduces diff-output if you would add another parameter.